computedfields/resolver.py

```diff
@@ -837,7 +837,30 @@ def _save(instance, *args, **kwargs):
         
         return wrap(func) if func else wrap
 
-    def update_computedfields(self, instance, update_fields=None):
+    def update_computedfields_on_parents(self, instance, model, update_fields=None):
+        """
+        Iterate over `instance`'s non-abstract parent models, if any, recursively
+        updating `instance`'s local computed fields that are contributed by its
+        non-abstract parent models
+
+        Returns ``None`` or an updated set of field names for `update_fields`.
+        The returned fields might contained additional computed fields, that also
+        changed based on the input fields, thus should extend `update_fields`
+        on a save call.
+        """
+        parents = model._meta.parents
+        if not parents:
+            return None
+
+        all_parent_fields = set()
+        for parent_model in parents.keys():
+            parent_fields = self.update_computedfields(instance, model=parent_model, update_fields=update_fields)
+            if parent_fields:
+                all_parent_fields.update(parent_fields)
+
+        return all_parent_fields
```

Comment:
Not sure here, if this correctly reflects in the MRO updating fields in the correct order. Furthermore, cfs living on its own parent table will have a different querystring later on, if some other field depends on them, which somewhat qualifies them to be handled as foreign model.

Reply:
mmmh yes I realize that I haven't tested other models CFs depending on a Parent model fields. Need to test both when the other model is related directly to the Parent model, and when it is related to a Child model. I also need to test the case of a Child's CF depending only on a Parent's field. I'll add those tests and see how it works, although I wouldn't be surprised that it fails